engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -1226,41 +1237,20 @@ private QueryTable whereInternal(final WhereFilter... filters) {
 
                                     final CompletableFuture<TrackingWritableRowSet> currentMappingFuture =
                                             new CompletableFuture<>();
+
                                     final InitialFilterExecution initialFilterExecution = new InitialFilterExecution(
-                                            this, filters, rowSetToUse.copy(), 0, rowSetToUse.size(), null, 0,
-                                            usePrev) {
-                                        @Override
-                                        void handleUncaughtException(Exception throwable) {
-                                            currentMappingFuture.completeExceptionally(throwable);
-                                        }
-                                    };
-                                    final ExecutionContext executionContext = ExecutionContext.getContext();
-                                    initialFilterExecution.scheduleCompletion(x -> {
-                                        try (final SafeCloseable ignored = executionContext.open()) {
-                                            if (x.exceptionResult != null) {
-                                                currentMappingFuture.completeExceptionally(x.exceptionResult);
-                                            } else {
-                                                currentMappingFuture.complete(x.addedResult.toTracking());
-                                            }
-                                        }
+                                            this, filters, rowSetToUse.copy(), usePrev);
+                                    final TrackingWritableRowSet currentMapping;
+                                    initialFilterExecution.scheduleCompletion(() -> {
+                                        currentMappingFuture.complete(initialFilterExecution.addedResult.toTracking());
+                                    }, exception -> {
```

Comment:
Cutely, you can pass in the method reference instead: `currentMappingFuture::completeExceptionally`